integration/spark-common/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala

```diff
@@ -191,6 +193,27 @@ object CarbonMergeFilesRDD {
     mergeIndexSize
   }
 
+  /**
+   * delete the file with retry
+   */
+  def deleteFileWithRetry(filePath: String, _retryTimes: Integer): Unit = {
+    var retryTimes = _retryTimes
+    while (!deleteFile(filePath) && retryTimes > 0) {
```

Comment:
why not use `deleteAllCarbonFilesOfDir` ?

Because line 175: you have to loop again to delete the directories. May be slow on s3